espnet2/enh/loss/wrappers/mixit_solver.py

```diff
@@ -0,0 +1,86 @@
+import itertools
+
+import torch
+
+from espnet2.enh.loss.criterions.abs_loss import AbsEnhLoss
+from espnet2.enh.loss.wrappers.abs_wrapper import AbsLossWrapper
+
+
+class MixITSolver(AbsLossWrapper):
+    def __init__(
+        self,
+        criterion: AbsEnhLoss,
+        weight=1.0,
+    ):
+        """Mixture Invariant Training Solver.
+
+        Args:
+            criterion (AbsEnhLoss): an instance of AbsEnhLoss
+            weight (float): weight (between 0 and 1) of current loss
+                for multi-task learning.
+        """
+        super().__init__()
+        self.criterion = criterion
+        self.weight = weight
+
+    @property
+    def type(self):
+        return "mixit"
+
+    def forward(self, ref, inf, others={}):
```

Comment:
can we include the constrained mixIT variant for Speech Enhancement ?

Reply:
for the others: see 4.2 Section in https://arxiv.org/pdf/2006.12701.pdf

Reply:
Cool. Perhaps it would be better to leave it later.